feat: Clarify & Improve Full Edit mode (#7274)

codealchemist007 opened the pull request:
Closes #7274

## Changes

This PR improves the "Full Edit mode" (power edit) by:

1. **Renamed "Labels & Certifications; Stores, Origins, Traceability codes, Country, Categories" to "Edit everything"** with a clearer subtitle ("Labels, stores, origins, and more"):
   - Updated the edit product page tile
   - Consistent naming across the UI

2. **Added footer shortcut** for quick access:
   - New "Edit everything" button in the customizable product page footer
   - Configurable in footer settings alongside other actions

3. **Added missing sections** to power edit:
   - **Basic details**: Product name, Brand names, Quantity
   - **Additional details**: Website
   - These fields now editable directly in power edit without needing separate pages

## Implementation Details

- Created new `SimpleInputPageHelper` classes for product name, quantity, and website
- Both entry points (edit page tile + footer button) open the same power edit view with 10 sections
- Updated save logic to merge changes per background task stamp (avoiding duplicate tasks for fields sharing the same stamp)
- All existing power edit sections (Labels, Stores, Origins, Country, etc) remain unchanged

## Testing

- Verified both "Edit everything" entry points work correctly
- Confirmed all 10 sections appear in power edit
- Tested save functionality with multiple fields
- UI validated on Android device(Android 15) 

## Screenshots

![Screenshot_2025-12-18-00-58-48-567_org openfoodfacts scanner](https://github.com/user-attachments/assets/512d0b8a-f934-4a0e-8e17-74f31877b693)

![Screenshot_2025-12-18-00-59-21-537_org openfoodfacts scanner](https://github.com/user-attachments/assets/377b02ba-5ae0-4ea3-a6f3-3355f3b6ec3e)

codealchemist007 committed:
feat: Clarify & Improve Full Edit mode (#7274)

codealchemist007 committed:
Resolve merge conflicts with develop

codealchemist007 commented:
Hey @teolemon, I've completed #7274 (Edit everything feature). Could you please review the PR when you get a chance? Thanks!

teolemon commented:
@codealchemist007 I'm confused by the second screenshot. 
The customizable bar is in the product page view mode, and the customization screen should more or less look like this: 
<img width="294" height="517" alt="image" src="https://github.com/user-attachments/assets/9d0e7127-54bc-4ced-8d07-a21dd7d66580" />

codealchemist007 committed:
fix: Add editEverything to default footer action order

codealchemist007 commented:
Hi @teolemon, fixed the issue—"Edit everything" wasn't included in the default footer action order as you said before. I've pushed a commit that adds it to the customization modal.

The button now appears in the footer settings between "Edit" and "Add a price" as expected. 

Thanks again!

![Screenshot_2025-12-22-18-36-55-876_org openfoodfacts scanner](https://github.com/user-attachments/assets/877a578b-2b50-4e1f-80c3-a24f0cbb9e38)

codealchemist007 committed:
fix: Format Dart files

codealchemist007 commented:
@teolemon this is just a quick follow‑up on this PR. I have already added ‘Edit everything’ to the footer customization, updated the screenshot and properly formatted those dart files as well which were causing the test failure. Let me know if you’d like me to update the branch against develop or adjust anything. Thanks!

teolemon commented:
@codealchemist007 can you remove the replicated translations to make your PR more easily reviewable ? We have an automated system that will take care of the replication when merging the PR.

teolemon commented:
@codealchemist007 can you also add screenshots of the edit view itself, with the additional widgets ?

teolemon commented:
also @codealchemist007 
```
   info • Type could be non-nullable • packages/smooth_app/lib/pages/product/simple_input/simple_input_page_helpers.dart:427:19 • unnecessary_nullable_for_final_variable_declarations
```

teolemon commented:
Asking Copilot for an initial review before pinging the team.

g123k commented:
Honestly, I'm not convinced by this fake duplication of the Edit mode.
The "Edit everything" misses the product name, photos…

Some alternatives
1. Create a wizard-like Edit mode (= a new Edit mode)
2. Add per field in the product page the ability to edit it directly

codealchemist007 committed:
chore: remove replicated translations and generated localization files

codealchemist007 committed:
refactor: deduplicate single-value helper logic and fix product name clearing

teolemon commented:
@codealchemist007 you have deleted the *.arb files

codealchemist007 committed:
fix: remove unnecessary nullable variable declarations in helpers

codealchemist007 commented:
> @codealchemist007 you have deleted the *.arb files

yeah I removed the non-English .arb and generated localization files as you requested earlier. If you’d like me to restore any of them, just let me know and I’ll do it right away!

teolemon commented:
@codealchemist007 we should keep the non-english arb files, it's only the app_localizations_**.dart I was talking about.
The idea is that the PR should be 5/6 files max, not 140

codealchemist007 commented:
> @codealchemist007 can you also add screenshots of the edit view itself, with the additional widgets ?

![Screenshot_2025-12-30-18-26-10-195_org openfoodfacts scanner](https://github.com/user-attachments/assets/cff17305-3f46-43fc-850a-52330f82f700)

![Screenshot_2025-12-30-18-26-15-055_org openfoodfacts scanner](https://github.com/user-attachments/assets/e5c9d840-5d05-422a-99da-6acbda118f7a)

![Screenshot_2025-12-30-18-26-41-162_org openfoodfacts scanner](https://github.com/user-attachments/assets/fdb6a65b-012b-4018-a2a1-48666b55d6fb)

![Screenshot_2025-12-30-18-26-47-437_org openfoodfacts scanner](https://github.com/user-attachments/assets/0b248094-1703-4be5-92bc-4d7c490a0279)

Screenshots: “Edit everything” (power edit) screen with new fields

* Shows all new widgets/fields: Product Names, Brand names, Quantity and weight, Website, Labels, Stores, Origins, Traceability codes, Country, and Categories.

* Demonstrates the updated UI and confirms the new sections are present as requested in #7274.

codealchemist007 commented:
> @codealchemist007 we should keep the non-english arb files, it's only the app_localizations_**.dart I was talking about. The idea is that the PR should be 5/6 files max, not 140

Thanks for clarifying! I misunderstood and deleted the non-English .arb files as well.
I’ll restore all the app_*.arb files (except the generated .dart ones) and make sure only the intended files are in the PR.
Sorry for the confusion—fixing it now!

codealchemist007 committed:
chore: restore non-English arb files as requested

codealchemist007 commented:
@teolemon, all non-English .arb files have been restored as requested. Only the generated .dart localization files are removed from the PR now. Let me know if there’s anything else to adjust. Thanks again!

codealchemist007 commented:
> Honestly, I'm not convinced by this fake duplication of the Edit mode. The "Edit everything" misses the product name, photos…
> 
> Some alternatives
> 
> 1. Create a wizard-like Edit mode (= a new Edit mode)
> 2. Add per field in the product page the ability to edit it directly

Hi @g123k, thanks for your input! I agree a true “Edit everything” or wizard mode would be a great improvement. For this PR, I’m sticking to the scope of #7274, but I’d be happy to help discuss or work on a more comprehensive edit mode in a future issue.

codealchemist007 commented:
Happy New Year! 🎊

Quick update: I've addressed all the feedback—

Restored non-English .arb files, removed only generated .dart files ✓
Added screenshots of the "Edit everything" screen ✓
Refactored duplicate code and fixed nullable variable warnings ✓
Resolved merge conflicts ✓

The PR should be ready for review now. Let me know if there's anything else to adjust. Thanks for your patience and guidance!

codealchemist007 commented:
@teolemon just checking in—is there anything else you'd like me to adjust on the PR, or any blockers on your end? I'm happy to make any changes needed. Thanks!

teolemon commented:
@g123k I see it as a net improvement, even though your points are valid.
I'd be in favor of merging if it's technically ok

codealchemist007 commented:
@teolemon @g123k just checking in to see if there’s any update or if there’s anything else I can do to help move this forward. Thanks!

teolemon commented:
@codealchemist007 can you fix the formatting on Formatted packages/smooth_app/lib/pages/product/simple_input/simple_input_page_helpers.dart ?

codealchemist007 commented:
@teolemon hope you're doing well. I'm still happy to make any final changes needed to get this merged. Looking forward to your feedback!

codealchemist007 committed:
chore(pr): remove l10n generated and arb file noise from PR

codealchemist007 committed:
chore(pr): align l10n app_arb and localizations with base

codealchemist007 committed:
chore(pr): drop l10n arb/localization noise from PR

codealchemist007 committed:
style: format simple_input_page_helpers.dart

codealchemist007 committed:
chore: sync [pubspec.lock] with upstream

codealchemist007 commented:
@teolemon quick update: I resolved the PR merge conflicts (including [pubspec.lock](vscode-file://vscode-app/c:/Users/DELL/AppData/Local/Programs/Microsoft%20VS%20Code/591199df40/resources/app/out/vs/code/electron-browser/workbench/workbench.html)), removed the localization noise files from the PR diff and applied the formatting fix requested for simple_input_page_helpers.dart. The branch is now up to date and pushed. Could you please take a final look whenever you get a chance?

codealchemist007 committed:
Merge remote-tracking branch 'upstream/develop' into feature/full-edit-mode-ui

codealchemist007 committed:
chore: sync lockfile

codealchemist007 commented:
@teolemon just following up on this PR. I’ve synced the branch with latest base, resolved the conflict and pushed the final lockfile update. The PR should now be up to date and ready for final review. Please let me know if you want any last changes.

codealchemist007 committed:
Merge remote-tracking branch 'upstream/develop' into feature/full-edit-mode-ui

codealchemist007 commented:
@teolemon hope you’re doing well. Just a quick follow-up on this PR — I’ve synced it with the latest base branch and pushed the updates, so it should be fully up to date now. Whenever you get a moment, please take a final look. Thanks!

codealchemist007 committed:
chore: resolve pubspec.lock conflict

codealchemist007 committed:
chore: regen pubspec.lock
